src/main/java/com/pretz/dyntreeserver/service/UserService.java

```diff
@@ -38,7 +39,7 @@ public void setSeed(String seed) {
 
     public String validateUser(User user) {
         if (!userRepo.findByName(user.getName()).isPresent()
-                || !passwordEncoder.matches(user.getPassword(), userRepo.getOne(user.getId()).getPassword())) {
+                || !passwordEncoder.matches(user.getPassword(), userRepo.findByName(user.getName()).get().getPassword())) {
```

Comment:
As you're already doing `userRepo.findByName(user.getName())` one line higher, I'd propose to save the result and then operate on it, e.g.:
```
User sthUser = userRepo.findByName(user.getName()).get();
if (user == null || passwordEncoder [...])
   ...
```

Reply:
Good point, I've thought about it myself but ultimately left it without refactoring out new local variable so that you would have more stuff to lambast :D I'll fix it now.

Reply:
[Fixed here](https://github.com/Elm0xz/dyn-tree-server/commit/6cf6b99d2b081bad70896c855330552baaf45e24)